File: src/types/api/ragRepository.ts
/**
 * RAG Repository API type definitions
 * Types for managing RAG repositories in the application
 */


export interface RAGRepository {
  id: string
  name: string
  description?: string
  url: string
  enabled: boolean
  requires_auth: boolean
  auth_token?: string
  priority: number
  created_at: string
  updated_at: string
}

export interface CreateRAGRepositoryRequest {
  name: string
  description?: string
  url: string
  enabled?: boolean
  requires_auth?: boolean
  auth_token?: string
  priority?: number
}

export interface UpdateRAGRepositoryRequest {
  name?: string
  description?: string
  url?: string
  enabled?: boolean
  requires_auth?: boolean
  auth_token?: string
  priority?: number
}

export interface RAGRepositoryListResponse {
  repositories: RAGRepository[]
  total: number
  page: number
  per_page: number
}

export interface RAGRepositoryConnectionTestResponse {
  success: boolean
  message: string
  available_databases_count?: number
}

export interface DownloadRAGDatabaseFromRepositoryRequest {
  repository_id: string
  database_id: string
  target_provider_id: string
  database_name?: string
  database_alias?: string
}

File: src-tauri/src/database/models/rag_repository.rs
use chrono::{DateTime, Utc};
use schemars::JsonSchema;
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
pub struct RAGRepository {
    pub id: Uuid,
    pub name: String,
    pub description: Option<String>,
    pub url: String,
    pub enabled: bool,
    pub requires_auth: bool,
    pub auth_token: Option<String>,
    pub priority: i32,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Serialize, Deserialize, JsonSchema)]
pub struct CreateRAGRepositoryRequest {
    pub name: String,
    pub description: Option<String>,
    pub url: String,
    pub enabled: Option<bool>,
    pub requires_auth: Option<bool>,
    pub auth_token: Option<String>,
    pub priority: Option<i32>,
}

#[derive(Debug, Serialize, Deserialize, JsonSchema)]
pub struct UpdateRAGRepositoryRequest {
    pub name: Option<String>,
    pub description: Option<String>,
    pub url: Option<String>,
    pub enabled: Option<bool>,
    pub requires_auth: Option<bool>,
    pub auth_token: Option<String>,
    pub priority: Option<i32>,
}

#[derive(Debug, Serialize, Deserialize, JsonSchema)]
pub struct RAGRepositoryListResponse {
    pub repositories: Vec<RAGRepository>,
    pub total: i64,
    pub page: i32,
    pub per_page: i32,
}

#[derive(Debug, Serialize, Deserialize, JsonSchema)]
pub struct RAGRepositoryConnectionTestResponse {
    pub success: bool,
    pub message: String,
    pub available_databases_count: Option<i32>,
}
